File: VRSampleApp/VRSampleApp/Classes/PresentationLayer/Modules/VideoPlayerModule/ViewModel/VideoPlayerViewModel.m
//
//  VideoPlayerViewModel.m
//  VRSampleApp
//
//  Copyright © 2017 Ooyala Inc. All rights reserved.
//

#import "VideoPlayerViewModel.h"
#import "VideoItem.h"

@interface VideoPlayerViewModel ()

extern NSString *kLogFileName;

@end


@implementation VideoPlayerViewModel

#pragma mark - Constants

NSString *kLogFileName = @"ooyalaDebugLog.log";

#pragma mark - Initialization

- (instancetype)initWithVideoItem:(VideoItem *)videoItem
                            pcode:(NSString *)pcode
                           domain:(NSString *)domain
                 andQAModeEnabled:(BOOL)QAModeEnabled {
  if (self = [super init]) {
    _videoItem = videoItem;
    _pcode = pcode;
    _domain = domain;
    _QAModeEnabled = QAModeEnabled;
  }
  
  [self removeLogFileWithName:kLogFileName];
  
  return self;
}

#pragma mark - Public functions

- (void)debugPrint:(NSString *)debugString {
  NSString *logFilePath = [self pathToLogFileWithName:kLogFileName];
  NSString *dump = @"";
  NSError *error;
  
  if ([NSFileManager.defaultManager fileExistsAtPath:logFilePath]) {
    dump = [NSString stringWithContentsOfFile:logFilePath encoding:NSUTF8StringEncoding error:&error];
  }
  
  dump = [NSString stringWithFormat:@"%@ :::::::::: %@", dump, debugString];
  // Write to the file
  [dump writeToFile:logFilePath atomically:YES encoding:NSUTF8StringEncoding error:&error];
  
  if (error) {
    NSLog(@"Failed writing to log file: %@, Error: ", error.localizedDescription);
  }
}

#pragma mark - Private functions

- (NSString *)pathToLogFileWithName:(NSString *)fileName {
  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
  NSString *documentsDirectory = [paths firstObject];
  
  return [documentsDirectory stringByAppendingPathComponent:fileName];
}

- (void)removeLogFileWithName:(NSString *)fileName {
  NSString *logFilePath = [self pathToLogFileWithName:fileName];
  NSFileManager *fileManager = NSFileManager.defaultManager;
  
  if ([fileManager fileExistsAtPath:logFilePath]) {
    [fileManager removeItemAtPath:logFilePath error:NULL];
  }
}

@end

File: SwiftVRSkinSampleApp/SwiftVRSkinSampleApp/Classes/PresentationLayer/Modules/VideoPlayerModule/ViewModel/VideoPlayerViewModel.swift
//
//  VideoPlayerViewModel.swift
//  VRSkinSampleApp
//
//  Copyright © 2017 Ooyala, Inc. All rights reserved.
//

class VideoPlayerViewModel {

  // MARK: - Public properties
  var videoItem: VideoItem
  var pcode: String
  var domain: String
  var QAModeEnabled: Bool

  // MARK: - Constants

  private enum Constants {
    static let logFileName = "ooyalaDebugLog.log"
  }

  // MARK: - Initialization

  init(videoItem: VideoItem, pcode: String, domain: String, QAModeEnabled: Bool) {
    self.videoItem     = videoItem
    self.pcode         = pcode
    self.domain        = domain
    self.QAModeEnabled = QAModeEnabled

    removeLogFile(named: Constants.logFileName)
  }

  // MARK: - Public functions
  func debugPrint(debugString: String) {
    guard let logFileURL = pathToLogFile(named: Constants.logFileName) else { return }
    var dump = ""

    do {
      if FileManager.default.fileExists(atPath: logFileURL.path) {
        dump = try String(contentsOf: logFileURL, encoding: .utf8)
      }

      dump = "\(dump) :::::::::: \(debugString)"

      // Write to the file
      try dump.write(to: logFileURL, atomically: true, encoding: .utf8)

    } catch {
      print("Failed writing to log file: \(logFileURL.path), Error: " + error.localizedDescription)
    }
  }

  // MARK: - Private functions
  private func pathToLogFile(named name: String) -> URL? {
    guard let documentsPathURL = FileManager.default.urls(for: .documentDirectory,
                                                          in: .userDomainMask).first else {
                                                            return nil
    }
    return documentsPathURL.appendingPathComponent(name)
  }

  private func removeLogFile(named name: String) {
    guard let logFileURL = pathToLogFile(named: name) else { return }

    if FileManager.default.fileExists(atPath: logFileURL.path) {
      try? FileManager.default.removeItem(at: logFileURL)
    }
  }
}
